### PR TITLE
Development vineet

### PR DESCRIPTION
**Pull request by manojnagar**

### CODE REVIEW

File: app/src/main/java/com/goupwards/checkcreditflow/presenter/EducationalQualificationPresenter.java [hunk 1]
```diff
@@ -71,14 +72,20 @@ public void onRequestError(RequestError error) {
 
   private boolean validateData() {
     boolean result = true;
+    String minYear = String.valueOf(1970);
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/presenter/EducationalQualificationPresenter.java]: why this required?  remove them

File: app/src/main/java/com/goupwards/checkcreditflow/presenter/EducationalQualificationPresenter.java [hunk 2]
```diff
@@ -71,14 +72,20 @@ public void onRequestError(RequestError error) {
...
+    String maxYear = String.valueOf(2017);
     String year = educationalView.getYearCompletion();
     if (TextUtils.isEmpty(educationalView.getYearCompletion())) {
       educationalView.setYearCompletionError("Completion year not filled");
       result = false;
     } else if (year.length() != 4) {
       educationalView.setYearCompletionError("Year should be four digit");
       result = false;
-    } else {
+    } else if (Integer.parseInt(year) < 1970 && Integer.parseInt(year) > 2017){
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/presenter/EducationalQualificationPresenter.java]: First cast values in varialbe as int year = Integer.parseInt(year) than changet year < 1970 && year > 2017 or etc

File: app/src/main/java/com/goupwards/checkcreditflow/view/EducationalQualificationFragment.java [hunk 3]
```diff
@@ -37,6 +40,29 @@ protected void initViews(View rootView) {
     passingYear = (TextInputEditText) rootView.findViewById(R.id.edu_passing_year);
     passingYearInputLayout = (TextInputLayout) rootView.findViewById(R.id.edu_passing_year_input_layout);
 
+    passingYear.addTextChangedListener(new TextWatcher() {
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/view/EducationalQualificationFragment.java]: Why this listener required?

File: app/src/main/java/com/goupwards/checkcreditflow/view/EducationalQualificationFragment.java [hunk 4]
```diff
@@ -37,6 +40,29 @@ protected void initViews(View rootView) {
...
+      @Override
+      public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+
+      }
+
+      @Override
+      public void afterTextChanged(Editable editable) {
+        int length = editable.length();
+        if ((yearStart > length) && (length < yearEnd)) {
+          Toast.makeText(getActivity(),"Correct year",Toast.LENGTH_LONG).show();
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/view/EducationalQualificationFragment.java]: why?

File: app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java [hunk 5]
```diff
@@ -60,26 +77,63 @@ public boolean onTouch(View v, MotionEvent event) {
...
+    workingerror = (TextInputLayout)rootView.findViewById(R.id.workingerroe);
+    workexperienceerror = (TextInputLayout)rootView.findViewById(R.id.yearmontherror);
+    spinnerOrganisationerror =(TextInputLayout)rootView.findViewById(R.id.spinnerOrganisationerror);
+    radiobuttonerror = (TextInputLayout)rootView.findViewById(R.id.radioButtonerror);
+
+
+    if (radiogroup.getCheckedRadioButtonId() == -1){
+      // No buttons are checked
+      //show error
+      setYearsError("");
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java]: show meaning ful error

File: app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java [hunk 6]
```diff
@@ -60,26 +77,63 @@ public boolean onTouch(View v, MotionEvent event) {
...
+    // TODO send this radioButtonId to url.
   }
 
+  private void setOrganisationError() {
+
+    if (check = false){
+      spinnerOrganisationerror.setError("Field Not Selected");
+    } else {
+
+      spinnerOrganisationerror.setError("");
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java]: show meaning full error, or remove setError method

File: app/src/main/java/com/goupwards/checkcreditflow/view/InputFilterMinMax.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package com.goupwards.checkcreditflow.view;
+
+import android.text.InputFilter;
+import android.text.Spanned;
+
+/**
+ * Created by Vineet on 2/3/2017.
+ */
+
+public class InputFilterMinMax implements InputFilter {
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/checkcreditflow/view/InputFilterMinMax.java]: What is use case of this?

File: app/src/main/java/com/goupwards/kycflow/presenter/DocumentUploadPresenter.java [hunk 8]
```diff
@@ -13,4 +13,7 @@
   public DocumentUploadPresenter(DocumentUploadContract.View documentUploadView) {
     this.documentUploadView = documentUploadView;
   }
+
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/kycflow/presenter/DocumentUploadPresenter.java]: remove extra lines

File: app/src/main/java/com/goupwards/kycflow/view/DocumentUploadFragment.java [hunk 9]
```diff
@@ -22,6 +45,123 @@ protected int getLayoutResourceId() {
...
+        intent.putExtra(Intent.EXTRA_MIME_TYPES, mimetypes);
+        startActivityForResult(intent, REQUEST_CODE_OPEN);
+      }
+    });
+
+    cameraimage.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View v) {
+        try {
+          Intent cameraIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/kycflow/view/DocumentUploadFragment.java]: From where you picked this code?

File: app/src/main/java/com/goupwards/kycflow/view/UserInfoFragment.java [hunk 10]
```diff
@@ -22,6 +36,82 @@ protected int getLayoutResourceId() {
...
+    }
+
+    @Override
+    public String getaltPhoneNumber() {
+        return altnumberEditText.getText().toString();
+    }
+
+    @Override
+    public void showDialogToInputVerficationCode() {
+        AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(getActivity());
```
Comment (manojnagar) [inline app/src/main/java/com/goupwards/kycflow/view/UserInfoFragment.java]: Create dialog fragment and resuse

File: app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java (hunk 6 above)
Comment (vinnyHero) [inline app/src/main/java/com/goupwards/checkcreditflow/view/EmploymentDetailFragment.java]: i removed the setOrganisationError() .

File: app/src/main/java/com/goupwards/checkcreditflow/view/InputFilterMinMax.java (hunk 7 above)
Comment (vinnyHero) [inline app/src/main/java/com/goupwards/checkcreditflow/view/InputFilterMinMax.java]: to restrict user to enter values in range

File: app/src/main/java/com/goupwards/kycflow/view/DocumentUploadFragment.java (hunk 9 above)
Comment (vinnyHero) [inline app/src/main/java/com/goupwards/kycflow/view/DocumentUploadFragment.java]: to start camera to click image and then crop .

**Pull request merged**